specs/protocol/jovian/predeploys.md

```diff
@@ -0,0 +1,349 @@
+# Predeploys
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [WETH9](#weth9)
+- [L1Block](#l1block)
+  - [Functions](#functions)
+    - [`setCustomGasToken`](#setcustomgastoken)
+- [L2CrossDomainMessenger](#l2crossdomainmessenger)
+- [L2StandardBridge](#l2standardbridge)
+- [SequencerFeeVault](#sequencerfeevault)
+- [BaseFeeVault](#basefeevault)
+- [L1FeeVault](#l1feevault)
+- [Native Asset Liquidity](#native-asset-liquidity)
+  - [Functions](#functions-1)
+    - [`deposit`](#deposit)
+    - [`withdraw`](#withdraw)
+    - [`fund`](#fund)
+  - [Events](#events)
+    - [`LiquidityDeposited`](#liquiditydeposited)
+    - [`LiquidityWithdrawn`](#liquiditywithdrawn)
+    - [`LiquidityFunded`](#liquidityfunded)
+  - [Invariants](#invariants)
+- [Liquidity Controller](#liquidity-controller)
+  - [Functions](#functions-2)
+    - [`authorizeMinter`](#authorizeminter)
+    - [`deauthorizeMinter`](#deauthorizeminter)
+    - [`mint`](#mint)
+    - [`burn`](#burn)
+    - [`gasPayingAssetName`](#gaspayingassetname)
+    - [`gasPayingAssetSymbol`](#gaspayingassetsymbol)
+  - [Events](#events-1)
+    - [`MinterAuthorized`](#minterauthorized)
+    - [`AssetsMinted`](#assetsminted)
+    - [`AssetsBurned`](#assetsburned)
+  - [Invariants](#invariants-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Name                 | Address                                    | Introduced | Deprecated | Proxied |
+| -------------------- | ------------------------------------------ | ---------- | ---------- | ------- |
+| NativeAssetLiquidity | 0x4200000000000000000000000000000000000029 | Jovian     | No         | Yes     |
+| LiquidityController  | 0x420000000000000000000000000000000000002A | Jovian     | No         | Yes     |
```

Comment:
What if we remove `NativeAssetLiquidity` and `LiquidityController`, and instead allow only the Portal contract to mint native assets on L2 if CGT is enabled? [Here](https://github.com/QuarkChain/optimism/pull/49) is an implementation following this model.

Reply:
This couples the minting logic to be on L1. A design goal was to enable the minting logic to be on L1 or L2